src/contracts/core/AllocationManager.sol

```diff
@@ -0,0 +1,552 @@
+        info.encumberedMagnitude = _encumberedMagnitude;
+        info.effectTimestamp = 0;
+        info.pendingDiff = 0;
+
+        // If the cleared change was a deallocation, update encumbered magnitude
+        if (mInfo.pendingDiff < 0) {
+            info.encumberedMagnitude = _addInt128(_encumberedMagnitude, mInfo.pendingDiff);
+        }
+
+        return info;
+    }
+
+    /// @notice Update the operator's magnitude info in storage and their encumbered magnitude.
+    function _updateMagnitudeInfo(
+        address operator,
+        IStrategy strategy,
+        bytes32 operatorSetKey,
+        PendingMagnitudeInfo memory info
+    ) internal {
+        _operatorMagnitudeInfo[operator][strategy][operatorSetKey] = MagnitudeInfo({
+            currentMagnitude: info.currentMagnitude,
+            pendingDiff: info.pendingDiff,
+            effectTimestamp: info.effectTimestamp
+        });
+
+        encumberedMagnitude[operator][strategy] = info.encumberedMagnitude;
+        emit EncumberedMagnitudeUpdated(operator, strategy, info.encumberedMagnitude);
+    }
+
+    function _calcDelta(uint64 currentMagnitude, uint64 newMagnitude) internal pure returns (int128) {
+        return int128(uint128(newMagnitude)) - int128(uint128(currentMagnitude));
+    }
+
+    function _addInt128(uint64 a, int128 b) internal pure returns (uint64) {
+        return uint64(uint128(int128(uint128(a)) + b));
+    }
+
+    /// @dev Returns an `OperatorSet` encoded into a 32-byte value.
+    /// @param operatorSet The `OperatorSet` to encode.
+    function _encodeOperatorSet(
+        OperatorSet memory operatorSet
+    ) internal pure returns (bytes32) {
+        return bytes32(abi.encodePacked(operatorSet.avs, uint96(operatorSet.operatorSetId)));
+    }
+
+    /// @dev Returns an `OperatorSet` decoded from an encoded 32-byte value.
+    /// @param encoded The encoded `OperatorSet` to decode.
+    /// @dev Assumes `encoded` is encoded via `_encodeOperatorSet(operatorSet)`.
+    function _decodeOperatorSet(
+        bytes32 encoded
+    ) internal pure returns (OperatorSet memory) {
+        return OperatorSet({
+            avs: address(uint160(uint256(encoded) >> 96)),
+            operatorSetId: uint32(uint256(encoded) & type(uint96).max)
+        });
+    }
+
+    /**
+     *
+     *                         VIEW FUNCTIONS
+     *
+     */
+
+    /// @inheritdoc IAllocationManager
+    function getAllocationInfo(
+        address operator,
+        IStrategy strategy
+    ) external view returns (OperatorSet[] memory, MagnitudeInfo[] memory) {
+        OperatorSet[] memory operatorSets =
+            avsDirectory.getOperatorSetsOfOperator({operator: operator, start: 0, length: type(uint256).max});
+        MagnitudeInfo[] memory infos = getAllocationInfo(operator, strategy, operatorSets);
+        return (operatorSets, infos);
+    }
+
+    /// @inheritdoc IAllocationManager
+    function getAllocationInfo(
```

Comment:
is it safe to say that a `MagnitudeInfo` represents / is the same as `AllocationInfo`? Reading this method I was expecting the return type to be `AllocationInfo`. (I don't know much about Solidity, so if that is non-sense, just ignore it). Thank you

Reply:
This naming is fixed/consistent here: https://github.com/Layr-Labs/eigenlayer-contracts/pull/860

Reply:
(`getAllocation` returns an `Allocation` struct)